index.js

```diff
@@ -0,0 +1,123 @@
+var DAYS = [
+    'SUN',
+    'MON',
+    'TUE',
+    'WED',
+    'THU',
+    'FRI',
+    'SAT'
+  ];
+  
+  var MONTHS = [
+    'JAN',
+    'FEB',
+    'MAR',
+    'APR',
+    'MAY',
+    'JUN',
+    'JUL',
+    'AUG',
+    'SEP',
+    'OCT',
+    'NOV',
+    'DEC'
+  ];
+  
+  var calendarDates = document.querySelectorAll('tbody td');
+  var currentDay = document.querySelectorAll('.current-day p');
+  
+  function displayCalendar(today) {
+    var month = today.getMonth();
+    var year = today.getFullYear();
+  
+    if (month !== nowMonth || year !== nowYear) {
+      today.setDate(1);
+    }
+  
+    var date = today.getDate();
+    var day = today.getDay();
+  
+    var lastDate = new Date(today.getFullYear(), today.getMonth() + 1, 0).getDate();
+  
+    var currentMonth = document.querySelector('.current-month');
+  
+    currentMonth.textContent = MONTHS[month] + ' ' + year;
+  
+    var dayIndex = day;
+  
+    if (month === nowMonth && year === nowYear) {
+      currentDay[0].textContent = DAYS[nowDay];
+      currentDay[1].textContent = nowDate;
+    } else {
+      currentDay[0].textContent = DAYS[day];
+      currentDay[1].textContent = 1;
+    }
+  
+    for (var i = 0; i < calendarDates.length; i++) {
+      calendarDates[i].textContent = '';
+    }
+  
+    for (var i = 1; i <= lastDate; i++) {
+      calendarDates[dayIndex].textContent = i;
+      calendarDates[dayIndex].style.color = 'black';
+  
+      if (i === date && year === nowYear && month === nowMonth) {
+        calendarDates[dayIndex].style.color = 'red';
+      }
+  
+      updateSelectedDay(dayIndex, day);
+  
+      dayIndex++;
+    }
+  }
+  
+  function updateSelectedDay(currentIndex, firstDay) {
```

Comment:
해당 함수는 단지 `calendarDates[currentIndex]`에 이벤트를 등록해주는 역할만하고 있습니다. 
위 함수명을 적절하게 변경하고(ex. `addEventListenerInDates`등), 
의도하신 `updateSelectedDay`는 날짜를 클릭했을 때 실행되는 콜백함수명으로 넣어주는 것이 더 자연스러울 것 같습니다.

```suggestion
  function addEventListenerInDates(currentIndex, firstDay) {
    function updateSelectedDay(ev) {
      if (!ev.target.textContent) {
        return;
      }

      if (currentIndex < 7) {
        currentDay[0].textContent = DAYS[currentIndex];
      } else {
        currentDay[0].textContent = DAYS[currentIndex % 7];
      }

      var newIndex = currentIndex - firstDay;
      currentDay[1].textContent = newIndex + 1;
    }

    calendarDates[currentIndex].addEventListener("click", updateSelectedDay);
  }
```